eZ/Publish/LegacySearch/LegacySearchFactory.php

```diff
@@ -8,17 +8,12 @@
 
 namespace EzSystems\RecommendationBundle\eZ\Publish\LegacySearch;
 
-/**
- * @TODO fixme
- */
+use EzSystems\RecommendationBundle\eZ\Publish\LegacySearch;
+
 class LegacySearchFactory
 {
     public static function build(\Closure $legacyKernelClosure)
     {
-        return $legacyKernelClosure()->runCallback(
```

Comment:
Hmm, no... :-)

The idea was to try/catch, and return the NullSearchEngine _IF_ an exception was raised. You're forcing the null storage engine in all cases, disabling search altogether in legacy. That's a bit harsh :-)